File: src/main/java/net/veloxia/logis/objects/LanguageData.java
package net.veloxia.logis.objects;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;

/**
 * Data class representing the language JSON file.
 *
 * @author xRookieFight
 * @since 23/03/2026
 */
@Data
@NoArgsConstructor
public class LanguageData {

    private String name;
    private String prettyName;
    private String flag;
    private List<String> translators;
    private Map<String, String> translations;
}

File: src/main/kotlin/net/veloxia/logis/objects/LanguageData.kt
package net.veloxia.logis.objects

import kotlinx.serialization.Serializable

/**
 * Data class representing the language JSON file.
 * @author xRookieFight
 */
@Serializable
data class LanguageData(
    val name: String,
    val prettyName: String,
    val flag: String,
    val translators: List<String>,
    val translations: Map<String, String>
) {
    override fun toString() : String = super.toString()
}
